src/ResourceGenerator.php

```diff
@@ -124,7 +124,7 @@ public function fromArray(array $data, string $uri = null) : HalResource
      *     against types registered in the metadata map.
      * @param ServerRequestInterface $request
      */
-    public function fromObject($instance, ServerRequestInterface $request) : HalResource
+    public function fromObject($instance, ServerRequestInterface $request, int $depth = 0) : HalResource
```

Comment:
This is a BC break. Classes extending this one and defining using the previous signature will produce a warning under PHP 7, and a fatal under PHP 8, due to signature mismatch. We'll need to target next major for this.